lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -193,6 +193,67 @@ def wire_target_dependencies(target_installation_results)
           PodTargetDependencyInstaller.new(sandbox, pod_target_installation_results_hash, metadata_cache).install!
         end
 
+        def _merge_spm_requirements(url, requirements)
+          if requirements.count > 1
+            requirements.reduce(requirements.first) do |acc, req|
+              if acc != req
+                error "SPM dependencies with different requirements: #{requirements} for #{url} is not yet supported"
+              end
+              acc
+            end
+          else
+            requirements.first
+          end
+        end
+
+        def install_spm_dedpendecies(project, pod_targets)
+          spm_dependencies = pod_targets.flat_map { |target| target.root_spec.consumer(target.platform).spm_dependencies }
```

Comment:
updating this line to get the dependencies on all the specs lets us define spm dependencies at the subspec level 🙂 

```suggestion
          spm_dependencies = pod_targets.flat_map { |target|
            target.specs.flat_map{ |spec| spec.consumer(target.platform).spm_dependencies }
          }
```